### ISSUE TITLE
mAP is Zero always

### ISSUE THREAD

**Issue by Thanossrs** [ultralytics/yolov5#1642]
## ❔Question
Could someone explain me:
1) I run custom yolov5 to Google Colab from Roboflow code.

2) and I follow step by step and the mAP is always zero and I do not see the boundary boxing after any prediction.

could someone help me?

thank you!!

**Comment by glenn-jocher** [comment]
@Thanossrs start from the Custom Training Tutorial, and once you have this working then swap in your custom dataset. See https://docs.ultralytics.com/yolov5/tutorials/train_custom_data

**Comment by ruyueshuo** [comment]
> ## ❔Question
> Could someone explain me:
> 
> 1. I run custom yolov5 to Google Colab from Roboflow code.
> 2. and I follow step by step and the mAP is always zero and I do not see the boundary boxing after any prediction.
> 
> could someone help me?
> 
> thank you!!
> 
> ## Additional context
@glenn-jocher 
Same problem. The training loss was still decreasing, but testing map was 0. I checked the inference outputs and the prob of candidates were all below 0.02. So they were removed by nms. What can I do ?

**Comment by ghost** [comment]
@ruyueshuo @Thanossrs Are you training from scratch on a small dataset like coco128? Try training from pretrained weights or training on a larger dataset.

**Comment by Thanossrs** [comment]
> @ruyueshuo @Thanossrs Are you training from scratch on a small dataset like coco128? Try training from pretrained weights or training on a larger dataset.

@PT0x00  thank you for your time.  hmm, I am not sure if I am training from scratch. I need to try from pretrained weights.
Could you help me or have you any helpful links?

**Comment by ghost** [comment]
> 
> 
> > @ruyueshuo @Thanossrs Are you training from scratch on a small dataset like coco128? Try training from pretrained weights or training on a larger dataset.
> 
> @PT0x00 thank you for your time. hmm, I am not sure if I am training from scratch. I need to try from pretrained weights.
> Could you help me or have you any helpful links?

If you set `--weights=' '` and `--resume=False` , you are probably training from scratch. You can pass supported weights to `--weights`, the code will automatically download them. Or you can download them manually from [releases](https://github.com/ultralytics/yolov5/releases).

**Comment by Thanossrs** [comment]
@PT0x00 I am trying and the mAP stills zero.
I change as you told me the --weights, but the mAP stills zero.
I need to do transfer learning and also I tried to freeze the layers but mAP stills zero.
Any idea?

**Comment by ghost** [comment]
> @PT0x00 I am trying and the mAP stills zero.
> I change as you told me the --weights, but the mAP stills zero.
> I need to do transfer learning and also I tried to freeze the layers but mAP stills zero.
> Any idea?

Sorry I can't figure out the exact problem with such information.

**Comment by cridonoso** [comment]
Hi @Thanossrs , I had the same problem. 
What I did was increase the number of images by subsampling with a smaller window of size W < Original Size. I also overlapped those windows.

I'm not sure if this was the problem, but now its working.
Best

**Comment by Thanossrs** [comment]
> Hi @Thanossrs , I had the same problem.
> What I did was increase the number of images by subsampling with a smaller window of size W < Original Size. I also overlapped those windows.
> 
> I'm not sure if this was the problem, but now its working.
> Best

@cridonoso Thank you for your reply I will try it!!

**Comment by Thanossrs** [comment]
> @Thanossrs start from the Custom Training Tutorial, and once you have this working then swap in your custom dataset. See https://docs.ultralytics.com/yolov5/tutorials/train_custom_data

@glenn-jocher thank you for your reply

**Comment by Thanossrs** [comment]
> > @PT0x00 I am trying and the mAP stills zero.
> > I change as you told me the --weights, but the mAP stills zero.
> > I need to do transfer learning and also I tried to freeze the layers but mAP stills zero.
> > Any idea?
> 
> Sorry I can't figure out the exact problem with such information.

**Comment by JagdishKolhe** [comment]
The issue is still there, Please refer the notebook here. 

https://nbviewer.jupyter.org/github/JagdishKolhe/odometer-reading/blob/main/Odopmeter_v1.ipynb

This training done on 16th Aug 2021. Please see the attached graph as well
![graph](https://user-images.githubusercontent.com/54310972/129762086-25a318fe-eff6-4dcc-9598-1e28550559ad.png)
